src/Microsoft.AspNetCore.Blazor.Build/Core/IndexHtmlWriter.cs

```diff
@@ -1,39 +1,40 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
-using Microsoft.AspNetCore.Blazor.Internal.Common.FileProviders;
+using System;
 using System.Collections.Generic;
-using System.Text;
-using Microsoft.Extensions.FileProviders;
+using System.IO;
 using System.Linq;
-using AngleSharp.Parser.Html;
+using System.Text;
 using AngleSharp;
 using AngleSharp.Html;
-using System;
+using AngleSharp.Parser.Html;
+using Mono.Cecil;
 
-namespace Microsoft.AspNetCore.Blazor.Build.Core.FileSystem
+namespace Microsoft.AspNetCore.Blazor.Build
 {
-    internal class IndexHtmlFileProvider : InMemoryFileProvider
+    public class IndexHtmlWriter
```

Comment:
Is there a reason why this isn't a `Task`? I'm not sure that I have the big picture totally about whether or not the tools needs to be standalone or integrated with MSBuild. If the latter, then just writing a task will simplify your life greatly.

Reply:
@rynowak Steve had all these things in a CLI tool and I want to keep it that way for now as this dll runs in a separate process from MSBuild and we get to have random dependencies. If I were to build an msbuild task for this it would just wrap the executable.

Reply:
> If I were to build an msbuild task for this it would just wrap the executable.

We made the decision to do this (wrap the executable) in Razor, and I don't regret it. I think it's fine for now.

Reply:
Looks good to me.